apollo-runtime/src/main/java/com/apollographql/apollo/ApolloMutationCall.java

```diff
@@ -40,9 +52,11 @@
    * @param requestHeaders The {@link RequestHeaders} to use for this request.
    * @return The ApolloCall object with the provided {@link RequestHeaders}.
    */
-  @NotNull ApolloMutationCall<T> requestHeaders(@NotNull RequestHeaders requestHeaders);
+  @Deprecated @NotNull ApolloMutationCall<T> requestHeaders(@NotNull RequestHeaders requestHeaders);
+
+  @Deprecated @NotNull @Override ApolloMutationCall<T> clone();
 
-  @NotNull @Override ApolloMutationCall<T> clone();
+  @NotNull @Override ApolloMutationCall.Builder<T> toBuilder();
```

Comment:
While reviewing I remembered I also added similar capability to `ApolloClient`. There I used `newBuilder` naming since OkHttp and Retrofit use that. I don't have a strong feeling but we should align them. 

https://github.com/apollographql/apollo-android/blob/master/apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java#L352

Reply:
I like `newBuilder()` better, it's more consistent with other libs. The question is whether we want to invest in this API if end up rebuilding it in `apollo-runtime-kotlin`. If we end up deprecating `apollo-runtime` in favor of `apollo-runtime-kotlin`, there's no need to deprecate too much.